packages/astro/src/core/config/schema.ts

```diff
@@ -149,7 +149,12 @@ export const AstroConfigSchema = z.object({
 			.optional()
 			.default({})
 	),
-	redirects: z.record(z.string(), z.string()).default(ASTRO_CONFIG_DEFAULTS.redirects),
+	redirects: z
+		.record(
+			z.string(),
+			z.union([z.string(), z.object({ status: z.number(), destination: z.string() })])
```

Comment:
Not sure if we need the extra strictness for allowed `status` numbers. I've made it only a `z.number()` for now.